usr/sbin/regionsrv-enabler-azure

```diff
@@ -96,21 +125,30 @@ def run_command(command):
 utils.start_logging()
 service_name = 'guestregister'
 license_type = get_license_type()
-
-if ('BYOS' in license_type and has_license_changed(license_type) and
-    not utils.uses_rmt_as_scc_proxy()):
+update_server = utils.get_smt()
+
+if ('BYOS' in license_type and not utils.uses_rmt_as_scc_proxy()):
+    license_has_changed = has_license_changed(license_type)
+    if (
+        license_has_changed or
+        (update_server and utils.is_registered(update_server.get_FQDN()))
+    ):
+        # either license has changed or
+        # system was previously registered
+        if license_has_changed:
+            update_license_cache(license_type)
```

Comment:
Running `--clean` here indiscriminately will bring a bug we fixed previously. The scenario is a user starting a BYOS instance, then registering it to SCC then using AHB to set he instance as BYOS.

In this case we will detect `BYOS` in the metadata and since the system is registered to SCC it will not use the update infrastructure as a proxy. Meaning we enter the code branch at line 130 and therefore we will wipe out the users SCC registration, which we should not do.

Reply:
I think I fixed both scenarios (the one we want to fix on this PR and the wonrgly de-registration if system is registered to SCC) in this commit [2bec9b4](https://github.com/SUSE-Enceladus/cloud-regionsrv-client/pull/138/commits/2bec9b4871e0bf41449a9930212b43368bf33cd0)

Reply:
Agreed with the addition of the scc check in the if condition on line 131 the issue I pointed out is resolved.